features/FEATURE_BLE/ble/BLE.h

```diff
@@ -421,1539 +421,18 @@ class BLE {
      */
     static const char *errorToString(ble_error_t error);
 
-    /*
-     * Deprecation alert!
-     * All of the following are deprecated and may be dropped in a future
-     * release. Documentation should refer to alternative APIs.
-     */
-public:
+private:
+    friend class BLEInstanceBase;
+
     /**
      * Constructor for a handle to a BLE instance (the BLE stack). BLE handles
      * are thin wrappers around a transport object (that is, ptr. to
      * BLEInstanceBase).
      *
      * @param[in] instanceID BLE Instance ID to get. 
-     *
-     * It is better to create BLE objects as singletons accessed through the
-     * Instance() method. If multiple BLE handles are constructed for the same
-     * interface (using this constructor), they share the same underlying
-     * transport object.
-     *
-     * @deprecated Use the Instance() function instead of the constructor.
      */
-    MBED_DEPRECATED("Use BLE::Instance() instead of BLE constructor.")
     BLE(InstanceID_t instanceID = DEFAULT_INSTANCE);
```

Comment:
Now the constructor is private as the internal stack still needs it.
(This comment forces GitHub to render large diff too)